inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/gather_elements.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+// //
+
+#include <vector>
+#include <ngraph/opsets/opset6.hpp>
+
+#include "single_layer_tests/gather_elements.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+using namespace ngraph::opset6;
+
+namespace {
+
+const std::vector<InferenceEngine::Precision> inputPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::FP16,
+        InferenceEngine::Precision::I32,
+};
+
+const std::vector<InferenceEngine::Precision> idxPrecisions = {
+        InferenceEngine::Precision::I32,
+        InferenceEngine::Precision::I64,
+};
+
+INSTANTIATE_TEST_CASE_P(smoke_set1, GatherElementsLayerTest,
+                        ::testing::Combine(
+                            ::testing::Values(std::vector<size_t>({2, 2})),     // Data shape
+                            ::testing::Values(std::vector<size_t>({2, 2})),     // Indices shape
+                            ::testing::ValuesIn(std::vector<int>({-1, 0, 1})),  // Axis
+                            ::testing::ValuesIn(inputPrecisions),
+                            ::testing::ValuesIn(idxPrecisions),
+                            ::testing::Values(CommonTestUtils::DEVICE_CPU)),
+                        GatherElementsLayerTest::getTestCaseName);
+
+INSTANTIATE_TEST_CASE_P(smoke_set2, GatherElementsLayerTest,
+                        ::testing::Combine(
+                            ::testing::Values(std::vector<size_t>({2, 2, 1})),  // Data shape
+                            ::testing::Values(std::vector<size_t>({4, 2, 1})),  // Indices shape
+                            ::testing::ValuesIn(std::vector<int>({0, -3})),     // Axis
+                            ::testing::ValuesIn(inputPrecisions),
+                            ::testing::ValuesIn(idxPrecisions),
+                            ::testing::Values(CommonTestUtils::DEVICE_CPU)),
+                        GatherElementsLayerTest::getTestCaseName);
+
+INSTANTIATE_TEST_CASE_P(smoke_set3, GatherElementsLayerTest,
+                        ::testing::Combine(
+                            ::testing::Values(std::vector<size_t>({2, 2, 3, 5})),   // Data shape
+                            ::testing::Values(std::vector<size_t>({2, 2, 3, 7})),   // Indices shape
+                            ::testing::Values(3, -1),                               // Axis
+                            ::testing::ValuesIn(inputPrecisions),
+                            ::testing::ValuesIn(idxPrecisions),
+                            ::testing::Values(CommonTestUtils::DEVICE_CPU)),
```

Comment:
are these CPU tests? if so, we dont need CPU tests under gpu test directory

Reply:
Yes, they are. I'll remove smoke_set1 ~ smoke_set5 cases from that file.

Reply:
what is the difference of this and the tests below?
Does it make sense just to keep these and chnage them to DEVICE_GPU?

Reply:
This part is from cpu functional test cases of gather elements, and below are the test cases I made. 
I forgot to modify the DEVICE_ part while bringing the existing test case to check that it also works on the gpu.
I'll change it to DEVICE_GPU and leave it if it works correctly.